openfermioncirq/gates/three_qubit_gates.py

```diff
@@ -112,25 +112,16 @@ def half_turns(self) -> Union[cirq.Symbol, float]:
         return self._exponent
 
     def _eigen_components(self):
-        return [
-            (0, numpy.diag([1, 1, 1, 1, 1, 0, 0, 1])),
-            (-0.5, numpy.array([[0, 0, 0, 0, 0, 0, 0, 0],
-                                [0, 0, 0, 0, 0, 0, 0, 0],
-                                [0, 0, 0, 0, 0, 0, 0, 0],
-                                [0, 0, 0, 0, 0, 0, 0, 0],
-                                [0, 0, 0, 0, 0, 0, 0, 0],
-                                [0, 0, 0, 0, 0, 0.5, 0.5, 0],
-                                [0, 0, 0, 0, 0, 0.5, 0.5, 0],
-                                [0, 0, 0, 0, 0, 0, 0, 0]])),
-            (0.5, numpy.array([[0, 0, 0, 0, 0, 0, 0, 0],
-                               [0, 0, 0, 0, 0, 0, 0, 0],
-                               [0, 0, 0, 0, 0, 0, 0, 0],
-                               [0, 0, 0, 0, 0, 0, 0, 0],
-                               [0, 0, 0, 0, 0, 0, 0, 0],
-                               [0, 0, 0, 0, 0, 0.5, -0.5, 0],
-                               [0, 0, 0, 0, 0, -0.5, 0.5, 0],
-                               [0, 0, 0, 0, 0, 0, 0, 0]]))
-        ]
+        minus_half_component = cirq.linalg.block_diag(
+            numpy.diag([0] * 5), numpy.array([[0.5, 0.5], [0.5, 0.5]]),
+            numpy.diag([0]))
+        plus_half_component = cirq.linalg.block_diag(
+            numpy.diag([0] * 5), numpy.array([[0.5, -0.5], [-0.5, 0.5]]),
+            numpy.diag([0]))
+
```

Comment:
Expand the zeros out and put each block on its own line.
```
        minus_half_component = cirq.linalg.block_diag(
            numpy.diag([0, 0, 0, 0, 0] ),
            numpy.array([[0.5, 0.5], [0.5, 0.5]]),
            numpy.diag([0]))

Reply:
Done - what's your tolerable limit for number of zeros before you'd convert to `numpy.diag([0] * n)`? I find these get hard to read pretty quickly (~5 is fine, but close to my limit).